components/select_team/components/select_team_item.tsx

```diff
@@ -12,21 +11,21 @@ import TeamInfoIcon from 'components/widgets/icons/team_info_icon';
 import {t} from 'utils/i18n';
 import * as Utils from 'utils/utils.jsx';
 
-export default class SelectTeamItem extends React.PureComponent {
-    static propTypes = {
-        team: PropTypes.object.isRequired,
-        onTeamClick: PropTypes.func.isRequired,
-        loading: PropTypes.bool.isRequired,
-        canJoinPublicTeams: PropTypes.bool.isRequired,
-        canJoinPrivateTeams: PropTypes.bool.isRequired,
-    };
+type Props = {
+    team: any,
+    onTeamClick: (team: any) => void,
```

Comment:
These can both use the `Team` type from `mattermost-redux/types/teams`. That can also be used in a few places in `SelectTeam` as well

Reply:
Thanks. Made this change.